File: Temp/leetcode/editor/cn/[997]找到小镇的法官.py
# 997 找到小镇的法官
# https://leetcode.cn/problems/find-the-town-judge/
from imports import *


# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def findJudge(self, n: int, trust: List[List[int]]) -> int:
        cnt, cnt_ed = defaultdict(int), defaultdict(int)
        for a, b in trust:
            cnt_ed[b] += 1  # b 被信任的数量 +1
            cnt[a] += 1  # a 信任的数量 +1
        for i in range(1, n + 1):
            if cnt[i] == 0 and cnt_ed[i] == n - 1:
                return i
        return -1
# leetcode submit region end(Prohibit modification and deletion)

File: Temp/leetcode/editor/cn/[997]找到小镇的法官.c
// 997 找到小镇的法官
// https://leetcode.cn/problems/find-the-town-judge/


//leetcode submit region begin(Prohibit modification and deletion)
int findJudge(int n, int** trust, int trustSize, int* trustColSize) {
    int* inDeg = (int*)malloc(sizeof(int) * (n + 1));
    memset(inDeg, 0, sizeof(int) * (n + 1));
    int* outDeg = (int*)malloc(sizeof(int) * (n + 1));
    memset(outDeg, 0, sizeof(int) * (n + 1));
    for (int i = 0; i < trustSize; i++) {
        int a = trust[i][0], b = trust[i][1];
        inDeg[b]++;
        outDeg[a]++;
    }
    for (int i = 1; i <= n; i++) {
        if (inDeg[i] == n - 1 && outDeg[i] == 0)
            return i;
    }
    free(inDeg);
    free(outDeg);
    return -1;
}
//leetcode submit region end(Prohibit modification and deletion)
